geomesa-memory/src/main/scala/org/locationtech/geomesa/memory/cqengine/utils/CQIndexingOptions.scala

```diff
@@ -0,0 +1,139 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.memory.cqengine.utils
+
+import java.util.UUID
+
+import com.googlecode.cqengine.attribute.Attribute
+import com.googlecode.cqengine.index.hash.HashIndex
+import com.googlecode.cqengine.index.radix.RadixTreeIndex
+import com.googlecode.cqengine.index.unique.UniqueIndex
+import com.googlecode.cqengine.{ConcurrentIndexedCollection, IndexedCollection}
+import com.typesafe.scalalogging.LazyLogging
+import com.vividsolutions.jts.geom.Geometry
+import org.locationtech.geomesa.memory.cqengine.attribute.SimpleFeatureAttribute
+import org.locationtech.geomesa.memory.cqengine.index.GeoIndex
+import org.locationtech.geomesa.memory.cqengine.utils.CQIndexType.CQIndexType
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes._
+import org.opengis.feature.`type`.AttributeDescriptor
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.collection.JavaConversions._
+import scala.util.Try
+
+// See geomesa/geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/Conversions.scala
+object CQIndexingOptions extends LazyLogging {
+  def getCQIndexType(ad: AttributeDescriptor): CQIndexType = {
+    Option(ad.getUserData.get(OPT_CQ_INDEX).asInstanceOf[String])
+      .flatMap(c => Try(CQIndexType.withName(c)).toOption).getOrElse(CQIndexType.NONE)
+  }
+
+  def setCQIndexType(ad: AttributeDescriptor, indexType: CQIndexType) {
+    ad.getUserData.put(OPT_CQ_INDEX, indexType.toString)
+  }
+
+  def buildIndexedCollection(sft: SimpleFeatureType): IndexedCollection[SimpleFeature] = {
+    val cqcache: IndexedCollection[SimpleFeature] = new ConcurrentIndexedCollection[SimpleFeature]()
+
+    // Add Geometry index on default geometry first.
+    addGeoIndex(sft.getGeometryDescriptor, cqcache)
+
+    // Add fid index
+    // TODO: should we always build this or expose building this as an option?
+    addFidIndex(cqcache)
```

Comment:
Actually, I thought we were using it in ``LiveFeatureCacheCQEngine.createOrUpdateFeature()`` but the Guava cache handles that.

Obviously it speeds up filters that contain by-id expressions, though only if it's logically joined with another expression. Filters with a single by-id expression currently get handled at the ``LiveFeatureCache`` level. Maybe I should remove.